public/controllers/agent/components/register-agent.js

```diff
@@ -233,20 +180,34 @@ export const RegisterAgent = withErrorBoundary(
         selectedVersion: '',
         selectedArchitecture: '',
-        selectedSYS: 'systemd',
+        selectedSYS: '',
       });
     }
 
     systemSelector() {
-      if (this.state.selectedOS === 'rpm') {
-        if (this.state.selectedSYS === 'systemd') {
-          return 'sudo systemctl daemon-reload\nsudo systemctl enable wazuh-agent\nsudo systemctl start wazuh-agent';
-        } else return 'sudo chkconfig --add wazuh-agent\nsudo service wazuh-agent start';
-      } else if (this.state.selectedOS === 'deb') {
-        if (this.state.selectedSYS === 'systemd') {
-          return 'sudo systemctl daemon-reload\nsudo systemctl enable wazuh-agent\nsudo systemctl start wazuh-agent';
-        } else return 'sudo update-rc.d wazuh-agent defaults 95 10\nsudo service wazuh-agent start';
-      } else return '';
+      if (this.state.selectedVersion === 'redhat7' || this.state.selectedVersion === 'amazonlinux2022' || this.state.selectedVersion === 'centos7' || this.state.selectedVersion === 'suse11' || this.state.selectedVersion === 'suse12' || this.state.selectedVersion === 'oraclelinux5' || this.state.selectedVersion === '22' || this.state.selectedVersion === 'amazonlinux2' || this.state.selectedVersion === 'debian8' || this.state.selectedVersion === 'debian9' || this.state.selectedVersion === 'debian10' || this.state.selectedVersion === 'busterorgreater' || this.state.selectedVersion === 'ubuntu15' || this.state.selectedVersion === 'ubuntu16' || this.state.selectedVersion === 'leap15') {
+        return 'sudo systemctl daemon-reload\nsudo systemctl enable wazuh-agent\nsudo systemctl start wazuh-agent';
+      } else if (this.state.selectedVersion === 'redhat5' || this.state.selectedVersion === 'redhat6' || this.state.selectedVersion === 'centos5' || this.state.selectedVersion === 'centos6' || this.state.selectedVersion === 'oraclelinux6' || this.state.selectedVersion === 'amazonlinux1' || this.state.selectedVersion === 'debian7' || this.state.selectedVersion === 'ubuntu14') {
+        return ('service wazuh-agent start')
+      }
+    }
+
+    systemSelectorNet() {
+      if (this.state.selectedVersion === 'windowsxp' || this.state.selectedVersion === 'windows8') {
+        return ('update-rc.d wazuh-agent defaults && service wazuh-agent start')
+      }
+    }
+
+    systemSelectorWazuhControlMacos() {
+      if (this.state.selectedVersion == 'sierra' || this.state.selectedVersion == 'highSierra' || this.state.selectedVersion == 'mojave' || this.state.selectedVersion == 'catalina' || this.state.selectedVersion == 'bigSur' || this.state.selectedVersion == 'monterrey') {
+        return ('/Library/Ossec/bin/wazuh-control start')
+      }
+    }
+
+    systemSelectorWazuhControl() {
+      if (this.state.selectedVersion === 'solaris10' || this.state.selectedVersion === 'solaris11' || this.state.selectedVersion === '6.1 TL9' || this.state.selectedVersion === '11.31') {
+        return ('/var/ossec/bin/wazuh-control start')
+      }
```

Comment:
Wouldn't it work to have an array with all the options and check if the selectedVersion is included in that array?

To make it easier to read and add or remove things.

Reply:
The development has been continued the way it was already, because the issue needed to be finished quickly.
The right thing to do would be to refactor the deploy new agent section, making clean, maintainable, scalable code.